Source/CombatExtended/CombatExtended/Jobs/JobDriver_ModifyWeapon.cs

```diff
@@ -89,12 +89,18 @@ public override IEnumerable<Toil> MakeNewToils()
             Toil waitToil = new Toil();
             waitToil.defaultDuration = 30;
             waitToil.defaultCompleteMode = ToilCompleteMode.Delay;
-            yield return waitToil.WithProgressBarToilDelay(TargetIndex.A);
+            yield return waitToil.WithProgressBarToilDelay(TargetIndex.A);            
             // modify the actual weapon            
             Toil modifyToil = new Toil();
             modifyToil.defaultCompleteMode = ToilCompleteMode.Instant;            
             modifyToil.initAction = () =>
-            {                
+            {
+                CompAmmoUser compAmmo = Weapon.TryGetComp<CompAmmoUser>();
+                if (compAmmo != null && compAmmo.HasMagazine && compAmmo.UseAmmo && !compAmmo.MagazineEmpty && !compAmmo.TryUnload(out _, forceUnload: true, pawn.Position))
+                {
+                    Log.Warning("CE: Failed to unload ammo from weapon before modifying it");
+                    Log.TryOpenLogWindow();
```

Comment:
Leftover from debugging?

Reply:
No, this one should be brought to the user's attention. It's not an error but it's not good either.